### DISCUSSION TITLE
Docker MeshCentral container does not start

### DISCUSSION THREAD

**Discussion by kmccmk9** [amidaware/tacticalrmm#1089] [General]
**Server Info (please complete the following information):**
 - OS: Ubuntu 20.04
 - Browser: Firefox
 - RMM Version: Latest

**Installation Method:**
  - [ ] Standard
  - [X] Docker

**Agent Info (please complete the following information):**
- Agent version: NA
- Agent OS: NA

**Describe the bug**
When running the docker compose file on a vanilla first time install, meshcentral container never starts up and repeatedly prints. "waiting for nginx to start...". Upon further inspection, this happens because the entrypoint.sh file has a hardcoded ip address that is not within the subnet created for this stack. Therefore the meshcentral container will never be able to communicate with the nginx container and this causes other containers that depend on meshcentral to fail as well.

**To Reproduce**
1. Download the docker compose file from this repository
2. Create a valid .env file from this repository
3. Start the compose file as documented
4. Check the logs for the init container and you will see it is waiting for the meshcentral container. 
5. Check the logs for the meshcentral container and you will see it can't connect to the nginx container.

**Expected behavior**
The containers should be on the same network or the entrypoint.sh should have a configurable environment variable to control the nginx IP

**Screenshots**
NA

**Additional context**
The current workaround is to acquire a shell of the running container and modify the entrypoint.sh file yourself.

**Comment by wh1te909** [comment]
I just tested a fresh docker install on ubuntu 20.04 on Hetzner cloud and it starts up just fine. Either you are not creating the `.env` file correctly or something is non standard with your setup, please provide more info on your setup.

**Reply by kmccmk9** [reply]
Hi, I deployed the compose file using  portainer. My environment variables are configured as such...
![image](https://user-images.githubusercontent.com/678335/166132331-e2668f89-cdd0-423f-9ba3-37981167dcb9.png)

I don't see anything out of place compared to the documentation?

**Reply by silversword411** [reply]
What isn't working?

**Reply by kmccmk9** [reply]
Meshcentral container never finishes initialization. It always prints "waiting for nginx to start..."

**Reply by kmccmk9** [reply]
@wh1te909 What other information can I provide?

**Reply by silversword411** [reply]
I wonder if there's a difference between docker and portainer. I thought portainer was just an enhanced management layer for docker but maybe not?

**Reply by ninjamonkey198206** [reply]
No difference between docker-compose and Portainer, as I use Portainer. I did a PR for the docker-compose file and env to add variables for the network settings due to random issues like this.

Setting the gateways, subnets, and IPs for important containers eliminates this problem.

**Reply by ninjamonkey198206** [reply]
When you create a stack in Portainer, it just stores docker-compose and env files that you filled in into a subfolder.

**Reply by ninjamonkey198206** [reply]
@wh1te909 it's an inconsistent issue, seemingly random. I've never found a pattern to it, only that it can occur when multiple networks are used in the same docker-compose/stack.

I've had the problem with other apps as well, so it's not unique to T-RMM.

**Reply by wh1te909** [reply]
@kmccmk9 @ninjamonkey198206 just another reason to not use docker. use standard install and you won't have these issues. docker is good for development and playing around with stuff, not for mission critical production systems. stop making your lives difficult lol.

**Reply by ninjamonkey198206** [reply]
It's a trade-off just like everything else.

It's perfectly fine and even preferable in production under certain circumstances.

I use baremetal or VMs for some services, Docker for others.
